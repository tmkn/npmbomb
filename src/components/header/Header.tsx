/** @jsx jsx */
import { jsx, css } from "@emotion/core";

import { mq } from "../../css";

const style = css({
    margin: "0 auto",
    "font-size": "2rem",
    "font-family": "'Muli', sans-serif",
    height: "4rem",
    display: "flex",
    alignItems: "center",
    [mq[0]]: {
        maxWidth: "300px"
    },
    [mq[1]]: {
        maxWidth: "700px"
    },
    [mq[2]]: {
        maxWidth: "900px"
    }
});

const titleStyle = css({
    flex: 1
});

const subTitleStyle = css({
    fontSize: "1rem",
    flex: 1,
    "font-family": "'Roboto Slab', serif",
    fontWeight: 400
});

const Header: React.FC = () => {
    return (
        <div css={style}>
            <div css={titleStyle}>npmbomb</div>
            <div css={subTitleStyle}>Guess NPM dependencies</div>
        </div>
    );
};

export default Header;
