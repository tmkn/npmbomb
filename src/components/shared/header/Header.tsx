/** @jsx jsx */
import { jsx, css } from "@emotion/core";

import { mq } from "../../../css";

const style = css({
    margin: "0 auto",
    fontSize: "2rem",
    fontFamily: "'Muli', sans-serif",
    height: "4rem",
    display: "flex",
    [mq[0]]: {
        maxWidth: "300px",
        flexDirection: "column",
        alignItems: "left",
        marginTop: "1rem"
    },
    [mq[1]]: {
        maxWidth: "700px",
        flexDirection: "row",
        alignItems: "center",
        marginTop: 0
    },
    [mq[2]]: {
        maxWidth: "900px"
    }
});

const titleStyle = css({
    flex: 1
});

const subTitleStyle = css({
    fontSize: "1rem",
    flex: 1,
    fontFamily: "'Roboto Slab', serif",
    fontWeight: 400
});

const Header: React.FC = () => {
    return (
        <div css={style}>
            <div css={titleStyle}>npmbomb</div>
            <div css={subTitleStyle}>Guess NPM dependencies</div>
        </div>
    );
};

export default Header;
