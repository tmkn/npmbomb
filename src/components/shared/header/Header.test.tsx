import React from "react";
import { BrowserRouter } from "react-router-dom";
import { render, fireEvent } from "@testing-library/react";

import Header from "./Header";

describe("<Header />", () => {
    test("matches snapshot", () => {
        const { asFragment } = render(
            <BrowserRouter>
                <Header />
            </BrowserRouter>
        );

        expect(asFragment()).toMatchSnapshot();
    });
});
