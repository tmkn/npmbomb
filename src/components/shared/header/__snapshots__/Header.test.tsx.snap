// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-denbe-Header"
  >
    <a
      href="/"
    >
      <div
        class="css-bof962-Header"
      >
        npmb
        <span
          class="css-1bxjuaj-Header"
        >
          ðŸ’£
        </span>
        mb
      </div>
    </a>
    <div
      class="css-fz30dm-Header"
    >
      Guess NPM dependencies
    </div>
  </div>
</DocumentFragment>
`;
